functions/functions.inc

```diff
@@ -442,3 +447,45 @@ function core_pages()
     return $valid;
 
 }
+
+
+function RBACApiRequest($rbac)
+{
+    global $_SESSION;
+
+    $target = '/permissions/meta/'.$rbac;
+
+    if (array_key_exists('API_SERVER', $_ENV)) {
+        $url = $_ENV['API_SERVER'].'/api/'.$target;
+    } elseif (isset($WEBSERVER)) {
+        $url = $WEBSERVER.'/api/'.$target;
+    } else {
+        $url = 'http://localhost//api/'.$target;
+        ;
+    }
+    $client = new Client();
+    try {
+        $request = new Request('GET', $url, ['Authorization' => 'Bearer '.$_SESSION['apiAuth']]);
+        $response = $client->send($request);
+    } catch (RequestException $e) {
+        $response = $e->getResponse();
+        error_log('API request failed: '.$response->getBody());
+        return null;
+    }
+    $data = json_decode($response->getBody());
+    return $data->allowed;
+
+}
+
+
+function PageAllowedByRBAC($rbac)
+{
+    $allowed = RBACApiRequest($rbac);
+    if ($allowed === null) {
+        LogoutUser();
+        goSite();
+    } elseif (!$allowed) {
+        goSite();
+    }
```

Comment:
Can you help me understand some of the decisions that have been made here? There is probably some context I don't have regarding the OAuth implementation.

- From what I gathered while poking around on my localhost, I noticed that we don't seem to make use of the refresh token portion of OAuth. Is there a specific reason we decided against this?
- I also noticed that, at least locally, the session tokens are set to expire after an hour. Is this just the default, or did we explicitly choose a 1 hour session time?

Now for the actual logic here:

- Will the user be notified when they are force-logged-out that their session had expired?
- Could this cause loss of work if someone's token expires before they're able to enter data?

Reply:
Ok this is now updated to handle renewals.... it is not ideal and maybe you know a better way to get that renewal to happen and then get the JavaScript data down into PHP.